src/EnergyPlus/Vectorize.hh

```diff
@@ -0,0 +1,34 @@
+#ifndef EnergyPlus_vectorize_hh_INCLUDED
+#define EnergyPlus_vectorize_hh_INCLUDED
+
+// EnergyPlus Vectorization Support
+
+// ObjexxFCL Headers
+#include <ObjexxFCL/vectorize.hh> // ASSUME_ALIGNED and RESTRICT macros
+
+// C++ Headers
+#include <cassert>
+#include <cstdint>
+
+// EnergyPlus headers
+#include "EnergyPlus.hh"
+
+namespace EnergyPlus {
+
+	int const VEC_LENGTH = VEC_ALIGN / sizeof(Real64);
+
+#define ROUND_TO_VEC_LENGTH(I) (((I) + ((VEC_LENGTH) - 1)) & ~((VEC_LENGTH) - 1))
+
+
+#define EXPLICIT_VECTORIZATION
+
+	// EXPLICIT_VECTORIZATION assumes that vectors are aligned,
+	// which is not a valid assumption on 32-bit builds
+#if defined(_WIN32) && !defined(_WIN64)
```

Comment:
The heap alignment is not fully specified by the C++ standard so we have to stay on top of the behavior for supported compilers, and we could break things for users building themselves on non-mainstream h/w or compilers. That said, I believe we are probably safe assuming 8 bytes on 32-bit builds and 16 bytes on 64-bit builds. I don't know of a macro-usable way to detect the heap alignment, but that would be the safest approach.

The ObjexxFCL can be built to align all the arrays to 32 bytes (AVX/2), 64 bytes (AVX-512/MIC), or higher. The current version support for this uses the non-standard and slow aligned allocator functions but the upcoming ObjexxFCL avoids those downsides. I'm also adding an aligned C-style array. And there are other vectorization-friendly vector libraries that can do specified heap alignment. So we should plan on exploiting at least AVX/2, if not in the initial release of this branch then as a follow up. It should give nearly double the speedup that the SSE version provides.